src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -130,6 +131,120 @@ internal ProcessOutputObject(object data, MinishellStream stream)
         }
     }
 
+    /// <summary>
+    /// This exception is used by the NativeCommandProcessor to indicate an error
+    /// when a native command retuns a non-zero exit code.
+    /// </summary>
+    [Serializable]
+    public class NativeCommandExitException : RuntimeException
```

Comment:
Since it is new class.
```suggestion
    [Serializable]
    #nullable enable
    public class NativeCommandExitException : RuntimeException
```